src/utils/sumo_utils.erl

```diff
@@ -33,7 +34,8 @@ doc_transform(Fun, Doc) ->
     FieldType = sumo_internal:field_type(Field),
     FieldName = sumo_internal:field_name(Field),
     FieldValue = sumo_internal:get_field(FieldName, Doc),
-    NewValue = Fun({FieldType, FieldName, FieldValue}),
+    FieldAttrs = sumo_internal:field_attrs(Field),
+    NewValue = Fun({FieldType, FieldName, FieldValue}, FieldAttrs),
```

Comment:
Why did you add `FieldAttrs` as an extra argument instead of adding it to the tuple?
Or rather… converting Fun in a 4-argument function?